storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -204,6 +205,29 @@ private boolean hasReorgedFrom(
     return Optional.empty();
   }
 
+  /**
+   * Returns the fork info that applies based on the node's current slot, regardless of where the
+   * sync progress is up to.
```

Comment:
This isn't quite true is it?  There could be an additional fork after the next fork (I assume the "next fork" is supposed to be the fork after the head state fork?) that is actually active for the current slot.  I assume at some point we'll have some explicit fork config that we can query ...  Might be worth a comment with caveats + javadoc on `getNextFork`.   And maybe `getCurrentForkInfo` should be `getHeadForkInfo`.